torch/quantization/fx/_equalize.py

```diff
@@ -220,3 +222,245 @@ def node_supports_equalization(node: Node, modules) -> bool:
     elif node.op == 'call_function':
         return node.target == nn.functional.linear
     return False
+
+def get_weight_eq_obs(
+    input_eq_obs_node: Node,
+    model: GraphModule,
+    modules: Dict[str, nn.Module]
+) -> Tuple[Optional[Node], Optional[_WeightEqualizationObserver]]:
+    """ Gets the following weight equalization observer. There should always
+    exsist a weight equalization observer after an input equalization observer.
+
+    Returns the node containing the weight equalization observer, and the weight
+    equalization observer if it has been newly created
+    """
+
+    # Find the next node that is either a nn.Linear or a functional layer
+    weight_node = input_eq_obs_node.next
+    while (weight_node.op not in ('output', 'call_function') and not
+           (weight_node.op == 'call_module' and isinstance(weight_node.target, str) and
+            isinstance(modules[weight_node.target], nn.Linear))):
+        weight_node = weight_node.next
+
+    if weight_node.op == 'call_module':
+        # If the next weight_node is a nn.Linear layer, then it must have a
+        # WeightEqualizationObserver configuration
+        equalization_qconfig_map: Dict[str, Any] = model._equalization_qconfig_map  # type: ignore[assignment]
+        assert(equalization_qconfig_map.get(weight_node.name, None) is not None)
+
+        weight_eq_obs = equalization_qconfig_map.get(weight_node.name, None).weight()
+        assert(isinstance(weight_eq_obs, _WeightEqualizationObserver))
+        return weight_node, weight_eq_obs
+
+    elif weight_node.op == 'call_function':
+        # TODO
+        return None, None
+
+    return None, None
+
+def get_next_input_obs(node: Node, modules: Dict[str, nn.Module]):
```

Comment:
@jerryzh168 do we currently have any helper functions that can return the input/output observer for a given node?

Reply:
is this what we are looking for? https://github.com/pytorch/pytorch/blob/master/torch/quantization/fx/quantization_patterns.py#L79

Reply:
I think the end goal of this function is to try to find the next input equalization observer given the current node. For example, if we have the given graph `linear1 -> out_obs1 -> eq_obs2 -> linear2` and `node = linear1`, then we would want to return the `eq_obs2`.

I might be reading the code wrong, but it seems like [maybe_get_last_node_only_observer](https://github.com/pytorch/pytorch/blob/master/torch/quantization/fx/quantization_patterns.py#L79) seems to just get the last observer node instead of finding the next observer node.

Reply:
oh, maybe_get_last_node_only_observer means get the only observer for the last node of the pattern. if we refactor the function a little bit, we can get the next observer node for a given node with this function, and looks like in this case we can call this function twice and get `eq_obs2`

Reply:
using `node.next` is a bit fragile I think, I think tracing the observer by finding the users of a given node would be more robust

Reply:
Ok! I will try that out with using `node.users.items()` then! I noticed that I actually only need to use this function to get the next input equalization observer. So do you think it would it be better if I wrote a `get_next_input_obs` function in [utils.py](https://github.com/pytorch/pytorch/blob/master/torch/quantization/fx/utils.py) and called this function twice to get the equalization observer, or would it be ok to just write a `get_next_input_eq_obs` function inside of this _equalize.py file?

Reply:
I didn't saw this code checking the type of module though, do we need that?

Reply:
I think we could extend https://github.com/pytorch/pytorch/blob/master/torch/quantization/fx/quantization_patterns.py#L79 to be
```python
def maybe_get_next_module(modules: Dict[str, torch.nn.Module], is_target_module_type):
```
where is_target_module_type is a function that checks if a module is the observer we are looking for or not.

Reply:
Ok, that makes sense to me. I'm not sure if I have access to a QuantizeHandler -- should I write this function in [utils.py](https://github.com/pytorch/pytorch/blob/master/torch/quantization/fx/utils.py) instead?

Reply:
yeah, that sounds good, we can update the callsite in QuantizeHandler to use this function as well in the future